source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1237 @@
+        {
+          a_dt = this->get_timestep()/year_in_seconds;
+        }
+
+      // We only want to run fastscape if there was a change in time.
+      if (a_dt > 0)
+        {
+          /*
+           * Initialize a vector of temporary variables to hold: z component, index, Vx, Vy, and Vz.
+           */
+          std::vector<std::vector<double>> temporary_variables(dim+2, std::vector<double>());
+          //Vector to hold the velocities that represent the change to the surface.
+          std::vector<double> V(array_size);
+          //precision = 0.001;
+
+          // Get a quadrature rule that exists only on the corners, and increase the refinement if specified.
+          const QIterated<dim-1> face_corners (QTrapez<1>(),
+                                               pow(2,additional_refinement+resolution_difference));
+
+          FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                            this->get_fe(),
+                                            face_corners,
+                                            update_values |
+                                            update_quadrature_points);
+
+          typename DoFHandler<dim>::active_cell_iterator
+          cell = this->get_dof_handler().begin_active(),
+          endc = this->get_dof_handler().end();
+
+          for (; cell != endc; ++cell)
+            if (cell->is_locally_owned() && cell->at_boundary())
+              for (unsigned int face_no = 0; face_no < GeometryInfo<dim>::faces_per_cell; ++face_no)
+                if (cell->face(face_no)->at_boundary())
+                  {
+                    if ( cell->face(face_no)->boundary_id() != relevant_boundary)
+                      continue;
+
+                    std::vector<Tensor<1,dim> > vel(face_corners.size());
+                    fe_face_values.reinit(cell, face_no);
+                    fe_face_values[this->introspection().extractors.velocities].get_function_values(this->get_solution(), vel);
+
+                    for (unsigned int corner = 0; corner < face_corners.size(); ++corner)
+                      {
+                        const Point<dim> vertex = fe_face_values.quadrature_point(corner);
+
+                        // Find what x point we're at. Add 1 or 2 depending on if ghost nodes are used.
+                        const double indx = 1+use_ghost+vertex(0)/dx;
+
+                        // If our x or y index isn't close to a whole number, then it's likely an artifact
+                        // from using an over-resolved quadrature rule, in that case ignore it.
+                        if (abs(indx - round(indx)) >= precision)
+                          continue;
+
+                        /*
+                         * If we're in 2D, we want to take the values and apply them to every row of X points.
+                         */
+                        if (dim == 2)
+                          {
+                            for (int ys=0; ys<ny; ys++)
```

Comment:
```suggestion
                            for (unsigned int ys=0; ys<ny; ys++)
```
?

Reply:
Ny and nx need to be regular int to work with fastscape, so I've left ys as int to avoid warnings.